Makefile

```diff
@@ -179,8 +182,54 @@ psql:
 shutdown-postgres:
 	@docker rm nexus-postgres --force
 
+# Used when RELEASE_VERSION is not set and until we have versioning tool in repo.
+NEXT_PATCH_VERSION := $(shell echo $(GIT_VERSION) | awk 'BEGIN{FS=OFS="."} {$$NF = $$NF + 1; print}')
+# Conditionally assign RELEASE_VERSION if it is not already set.
+RELEASE_VERSION ?= $(NEXT_PATCH_VERSION)
+# Git tag of the next release.
+RELEASE_TAG := v$(RELEASE_VERSION)
+
+confirm-version: fetch-git
+	@$(ECHO) "Latest published version is $(RED)v$(GIT_VERSION)$(OFF). You are about to publish $(CYAN)$(RELEASE_TAG)$(OFF)."
+	@$(ECHO) "If this is not what you want, re-run this command with RELEASE_VERSION=..." 
+	@$(CONFIRM_ACTION)
+
+# Fetch all the latest changes (including tags) from the canonical upstream git
+# repository.
+fetch-git:
+	@$(ECHO) "Fetching the latest changes (including tags) from $(GIT_ORIGIN_REMOTE) remote..."
+	@git fetch $(GIT_ORIGIN_REMOTE) --tags
+
+# Used when RELEASE_VERSION is not set and until we have versioning tool in repo.
+NEXT_PATCH_VERSION := $(shell echo $(GIT_VERSION) | awk 'BEGIN{FS=OFS="."} {$$NF = $$NF + 1; print}')
+# Conditionally assign RELEASE_VERSION if it is not already set.
+RELEASE_VERSION ?= $(NEXT_PATCH_VERSION)
+# Git tag of the next release.
+RELEASE_TAG := v$(RELEASE_VERSION)
+
```

Comment:
I think this is not needed because variable assignments apply to the whole Makefile. E.g. here, all three targets print "foom" in my testing:
```
one:
	@echo $(FOO)

FOO := foom

two:
	@echo $(FOO)

three:
	@echo $(FOO)
```

Reply:
it's not, made a mistake when I was moving stuff between commits